integration_test/tests/helper/Expect.ts

```diff
@@ -0,0 +1,208 @@
+        throwErr(errorMessage);
+      }
+    },
+
+    toBeTruthy() {
+      if (actual) logPass(`${actual} is truthy`);
+      else {
+        const errorMessage = `Expected ${actual} to be truthy`;
+        console.error(`[EXPECTATION NOT MET]: ${errorMessage}`);
+        throwErr(errorMessage);
+      }
+    },
+
+    toBeFalsy() {
+      if (!actual) logPass(`${actual} is falsy`);
+      else {
+        const errorMessage = `Expected ${actual} to be falsy`;
+        console.error(`[EXPECTATION NOT MET]: ${errorMessage}`);
+        throwErr(errorMessage);
+      }
+    },
+
+    toBeDefined() {
+      if (actual !== undefined) logPass(`${actual} is defined`);
+      else {
+        const errorMessage = `Expected ${actual} to be defined`;
+        console.error(`[EXPECTATION NOT MET]: ${errorMessage}`);
+        throwErr(errorMessage);
+      }
+    },
+
+    toBeUndefined() {
+      if (actual === undefined) logPass(`${actual} is undefined`);
+      else {
+        const errorMessage = `Expected ${actual} to be undefined`;
+        console.error(`[EXPECTATION NOT MET]: ${errorMessage}`);
+        throwErr(errorMessage);
+      }
+    },
+
+    toBeNull() {
+      if (actual === null) logPass(`${actual} is null`);
+      else {
+        const errorMessage = `Expected ${actual} to be null`;
+        console.error(`[EXPECTATION NOT MET]: ${errorMessage}`);
+        throwErr(errorMessage);
+      }
+    },
+
+    toBeNotNull() {
+      if (actual !== null) logPass(`${actual} is not null`);
+      else {
+        const errorMessage = `Expected ${actual} to be not null`;
+        console.error(`[EXPECTATION NOT MET]: ${errorMessage}`);
+        throwErr(errorMessage);
+      }
+    },
+
+    toBeInstanceOf(expected: any) {
+      if (actual instanceof expected)
+        logPass(`${actual} is instance of ${expected.name}`);
+      else {
+        const errorMessage = `Expected ${actual} to be instance of ${expected.name}`;
+        console.error(`[EXPECTATION NOT MET]: ${errorMessage}`);
+        throwErr(errorMessage);
+      }
+    },
+
+    toContain(expected: any) {
+      if (actual && actual.includes && actual.includes(expected))
+        logPass(`${actual} contains ${expected}`);
+      else {
+        const errorMessage = `Expected ${actual} to contain ${expected}`;
+        console.error(`[EXPECTATION NOT MET]: ${errorMessage}`);
+        throwErr(errorMessage);
+      }
+    },
+
+    toHaveLength(expected: number) {
+      if (actual && actual.length === expected)
+        logPass(`${actual} has length ${expected}`);
+      else {
+        const errorMessage = `Expected ${actual} to have length ${expected}, but got ${
+          actual ? actual.length : 'undefined'
+        }`;
+        console.error(`[EXPECTATION NOT MET]: ${errorMessage}`);
+        throwErr(errorMessage);
+      }
+    },
+
+    toHaveProperty(property: string) {
+      if (actual && actual.hasOwnProperty(property))
```

Comment:
Can we go through the object's prototype so that we are sure we're calling the native `hasOwnProperty`, only in case it exists?

Applies to `toHavePropertyValue` as well.

```suggestion
      if (actual && Object.prototype.hasOwnProperty.call(actual, property))
```

---

Otherwise this can fail in edge cases like:
```ts
const noProto = Object.create(null);
noProto.key = 123;
console.log(noProto.hasOwnProperty('key')); // Error: noProto.hasOwnProperty is not a function

const overridden = {
  key: 123,
  hasOwnProperty: () => false,
};
console.log(overridden.hasOwnProperty('key')); // false

console.log(Object.prototype.hasOwnProperty.call(overridden, 'key')) // true
console.log(Object.prototype.hasOwnProperty.call(noProto, 'key'))    // true
```